### PR TITLE
magento/devdocs#: Set page layout type

### PR DESCRIPTION
**Pull request by atwixfirster**
## Purpose of this pull request

This pull request (PR) extends `Page layout` > `Page layouts declaration` section. It adds information how developer can change page layout type.

## Affected DevDocs pages

- https://devdocs.magento.com/guides/v2.1/frontend-dev-guide/layouts/layout-types.html
- https://devdocs.magento.com/guides/v2.2/frontend-dev-guide/layouts/layout-types.html
- https://devdocs.magento.com/guides/v2.3/frontend-dev-guide/layouts/layout-types.html

## Links to Magento source code

- https://github.com/magento/magento2/blob/2.3.1/app/code/Magento/Theme/view/base/layouts.xml
- https://github.com/magento/magento2/blob/2.3.1/app/code/Magento/Theme/view/frontend/layouts.xml
- https://github.com/magento/magento2/blob/2.3.1/app/code/Magento/Theme/view/adminhtml/layouts.xml

whatsnew
Added information about [page layout types](https://devdocs.magento.com/guides/v2.3/frontend-dev-guide/layouts/layout-types.html) and a new code example that shows how to use them

### CODE REVIEW

**Comment by devops-devdocs** [comment]
An admin must run tests on this PR before it can be merged.

**Comment by atwixfirster** [comment]
> Hi @atwixfirster
> Thanks for your contributions. This PR is very helpful to the developer for understanding the `Page layout`.

Thanks for approve, @ronak2ram !

With my pleasure 😊

File: guides/v2.1/frontend-dev-guide/layouts/layout-types.md [hunk 1]
```diff
@@ -86,6 +86,31 @@ Sample page layout declaration file: `<Magento_Theme_module_dir>/view/frontend/l
 </page_layouts>
 ```
 
+Use `layout` attribute in `page` section to define a layout type of the page. Example below shows how to use `3 columns` page layout type for the [Wish List Sharing](https://github.com/magento/magento2/blob/2.3.1/app/code/Magento/Wishlist/view/frontend/layout/wishlist_index_share.xml#L8) page:
```
Comment (jeff-matthews) [inline guides/v2.1/frontend-dev-guide/layouts/layout-types.md]:
```suggestion
Use the `layout` attribute in the `page` section to define a layout type for the page. The following example shows how to use the `3 columns` page layout type for the [Wish List Sharing](https://github.com/magento/magento2/blob/2.3.1/app/code/Magento/Wishlist/view/frontend/layout/wishlist_index_share.xml#L8) page:
```

File: guides/v2.1/frontend-dev-guide/layouts/layout-types.md [hunk 2]
```diff
@@ -86,6 +86,31 @@ Sample page layout declaration file: `<Magento_Theme_module_dir>/view/frontend/l
...
+            <block class="Magento\Wishlist\Block\Customer\Sharing" name="wishlist.sharing" template="Magento_Wishlist::sharing.phtml" cacheable="false"/>
+        </referenceContainer>
+    </body>
+</page>
+```
+
+![Wish List Sharing. 3 columns layout page type]({{ page.baseurl }}/frontend-dev-guide/images/version.png)
+
+{:.bs-callout .bs-callout-info}
+By default Magento introduces 5 page layout types: `empty`, `1column`, `2columns-left`, `2columns-right` and `3columns` for frontend and 3 page layout types `admin-empty`, `admin-1column` and `admin-2columns-left` for backend.
```
Comment (jeff-matthews) [inline guides/v2.1/frontend-dev-guide/layouts/layout-types.md]:
```suggestion
By default, Magento provides 5 page layout types for the frontend (`empty`, `1column`, `2columns-left`, `2columns-right`, and `3columns`) and 3 page layout types for the backend (`admin-empty`, `admin-1column`, and `admin-2columns-left`).
```

Comment (atwixfirster) [reply guides/v2.1/frontend-dev-guide/layouts/layout-types.md]: **implemented**

Thank you!

File: guides/v2.1/frontend-dev-guide/layouts/layout-types.md (hunk 1 above)
Comment (atwixfirster) [reply guides/v2.1/frontend-dev-guide/layouts/layout-types.md]: **fixed**

**Comment by jeff-matthews** [comment]
running tests

**Comment by jeff-matthews** [comment]
Tests failed. Here's the log:

```
+ rake test:cicd
[35mChecking links with htmlproofer...[0m
Running ["LinkChecker::DoubleSlashCheck", "ImageCheck", "LinkCheck", "ScriptCheck"] on ["_site"] on *.html... 

Ran on 4105 files!

- internal image /guides/v2.1/frontend-dev-guide/images/version.png does not exist
  *  _site/guides/v2.1/frontend-dev-guide/layouts/layout-types.html (line 315)
- internal image /guides/v2.2/frontend-dev-guide/images/version.png does not exist
  *  _site/guides/v2.2/frontend-dev-guide/layouts/layout-types.html (line 315)
- internal image /guides/v2.3/frontend-dev-guide/images/version.png does not exist
  *  _site/guides/v2.3/frontend-dev-guide/layouts/layout-types.html (line 315)
rake aborted!
HTML-Proofer found 3 failures!```

File: guides/v2.1/frontend-dev-guide/layouts/layout-types.md [hunk 3]
```diff
@@ -86,6 +86,31 @@ Sample page layout declaration file: `<Magento_Theme_module_dir>/view/frontend/l
...
+    <update handle="customer_account"/>
+    <body>
+        <referenceContainer name="content">
+            <block class="Magento\Wishlist\Block\Customer\Sharing" name="wishlist.sharing" template="Magento_Wishlist::sharing.phtml" cacheable="false"/>
+        </referenceContainer>
+    </body>
+</page>
+```
+
+![Wish List Sharing. 3 columns layout page type]({{ page.baseurl }}/frontend-dev-guide/images/version.png)
```
Comment (jeff-matthews) [inline guides/v2.1/frontend-dev-guide/layouts/layout-types.md]: This only works if you add symlinks from 2.3 and 2.2 to the 2.1 image file you added.

Comment (atwixfirster) [reply guides/v2.1/frontend-dev-guide/layouts/layout-types.md]: I used a wrong file name. Could you please run tests again?

Thank you!

Comment (jeff-matthews) [reply guides/v2.1/frontend-dev-guide/layouts/layout-types.md]: Ah, yes. You're right. I didn't look closely enough at the errors. I'll run again soon.

Comment (atwixfirster) [reply guides/v2.1/frontend-dev-guide/layouts/layout-types.md]: > This only works if you add symlinks from 2.3 and 2.2 to the 2.1 image file you added.
Also, I've added symlinks ;)

**Comment by jeff-matthews** [comment]
running tests

**Pull request merged**